onnxruntime/core/session/environment.cc

```diff
@@ -66,6 +66,67 @@ Status Environment::RegisterAllocator(AllocatorPtr allocator) {
   return Status::OK();
 }
 
+Status Environment::CreateAndRegisterAlloctor(const OrtMemoryInfo& mem_info, const OrtArenaCfg* arena_cfg) {
+  // TODO should we allow sharing of non-CPU allocators?
+  if (mem_info.device.Type() != OrtDevice::CPU) {
```

Comment:
copied from the OrtEnv class to the intenral Environment class so that it can be shared between all language bindings